### PR TITLE
1257 introduce axiom model

### LINKED ISSUE #1257: Introduce Axiom model
Currently, we have the models `Sentence` and `Theorem < Sentence`. To get a better distinction, we want to have `Axiom < Sentence`, which contains all Sentences that are not Theorems.

Update the views to show Axioms instead of Sentences.

Remove the default scope of Sentence.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #1257. This pull request is based on the branch of #1188 (**1131-prove_theorems**) instead of **staging** . It might be easier to review that one first.

### CODE REVIEW

File: app/controllers/axioms_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        super
+      end
+  end
+
+  def logically_translated?
+    parent.contains_logic_translations?
+  end
+
+  helper_method :logically_translated?
+end
```
Comment (tillmo) [inline app/controllers/axioms_controller.rb]: acutally, axioms_controller.rb and sentences_controller.rb are identical, except that "axiom" is replaced by "sentence" at 5 places. Couldn't this be abstracted somehow?

File: app/models/ontology.rb [hunk 2]
```diff
@@ -158,6 +158,12 @@ def all_sentences
       where('imported = ? OR imported = ?', true, false)
   end
 
+  def all_axioms
+    Axiom.unscoped.
+      where(ontology_id: self).
+      where('imported = ? OR imported = ?', true, false)
```
Comment (tillmo) [inline app/models/ontology.rb]: what is the reason of this `where` clause, given that `imported boolean DEFAULT false NOT NULL`? Could `imported` be anything else than `true` or `false`?

File: app/controllers/axioms_controller.rb (hunk 1 above)
Comment (eugenk) [inline app/controllers/axioms_controller.rb]: Follow-up question: Do we want to have a SentencesController for the web app (i.e. not the API) serving both, axioms and theorems? If not, I would like to remove it.

File: app/models/ontology.rb (hunk 2 above)
Comment (eugenk) [inline app/models/ontology.rb]: Fixed with ae691abf44f1d6477ddb8bf5fd7e3b32457f3b7a

File: app/serializers/axiom_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class AxiomSerializer < ApplicationSerializer
+  class Reference < ApplicationSerializer
+    attributes :iri
+    attributes :name
+
+    def iri
+      qualified_locid_for(object)
+    end
+  end
```
Comment (tillmo) [inline app/serializers/axiom_serializer.rb]: could the need for this class Reference (which occurs in all the serializers) please be explained somewhere?

File: spec/models/ontology_spec.rb [hunk 4]
```diff
@@ -339,6 +339,26 @@
...
       end
     end
+
+    context 'axioms count' do
+      it 'should be correct' do
+        expect(ontology.axioms.count).to eq(1)
+      end
+
+      it 'should be reflected in the corresponding field' do
+        expect(ontology.axioms_count).to eq(ontology.axioms.count)
```
Comment (tillmo) [inline spec/models/ontology_spec.rb]: why do you test something to be equal to itself? What does this test really test? non-nil?

File: app/serializers/axiom_serializer.rb (hunk 3 above)
Comment (eugenk) [inline app/serializers/axiom_serializer.rb]: When the API is called e.g. for an ontology, we don't want to include all the versions in the response. Instead, we send a reference (IRI) to an index of the versions. This index in turn contains only references to the versions themselves.

This does not only apply to versions, but also for all the other association ends our models have. In order to have a consistent way of describing and serializing those references, we have those Reference classes everywhere. See for example the line `has_one :ontology, serializer: OntologySerializer::Reference ` below.

File: spec/models/ontology_spec.rb (hunk 4 above)
Comment (eugenk) [inline spec/models/ontology_spec.rb]: This is not quite the same. On the left hand side, we have `axioms_count` (underscore) which is a column in the database and on the right hand side we have `axioms.count` (dot) which actually sends an sql query to count the rows.

File: app/controllers/axioms_controller.rb (hunk 1 above)
Comment (tillmo) [inline app/controllers/axioms_controller.rb]: I think we don't need a SentencesController for serving sentences routes. We might need one if we want to abstract from both AxiomsController and TheoremsController. However, currently this seems not be necessary.

Comment (eugenk) [inline app/controllers/axioms_controller.rb]: Alright, then I'll remove the SentencesController, its routes and its views. I won't touch the API::SentencesController.

File: app/serializers/axiom_serializer.rb (hunk 3 above)
Comment (tillmo) [inline app/serializers/axiom_serializer.rb]: good. Could this explanation please be added in the code (or the doc) somewhere?

File: spec/models/ontology_spec.rb (hunk 4 above)
Comment (tillmo) [inline spec/models/ontology_spec.rb]: oh, sorry, I have overlooked this subtle difference.

**Comment by tillmo** [comment]
Since I do not get notified about the commits, a short message would be helpful. 

Everything looks good, but one point is still open: "Could this explanation please be added in the code (or the doc) somewhere?"

**Comment by eugenk** [comment]
> Could this explanation please be added in the code (or the doc) somewhere?

I have just completed it in #1288.

**Comment by tillmo** [comment]
:+1:

**Pull request merged**